pylearn2/datasets/csv_dataset.py

```diff
@@ -142,32 +151,35 @@ def _load_data(self):
         else:
             data = np.loadtxt(self.path, delimiter=self.delimiter)
 
-        def take_subset(X, y):
+        def take_subset(x, y):
             if self.start_fraction is not None:
-                n = X.shape[0]
+                n = x.shape[0]
                 subset_end = int(self.start_fraction * n)
-                X = X[0:subset_end, :]
+                x = x[0:subset_end, :]
                 y = y[0:subset_end]
             elif self.end_fraction is not None:
-                n = X.shape[0]
+                n = x.shape[0]
                 subset_start = int((1 - self.end_fraction) * n)
-                X = X[subset_start:, ]
+                x = x[subset_start:, ]
                 y = y[subset_start:]
             elif self.start is not None:
-                X = X[self.start:self.stop, ]
+                x = x[self.start:self.stop, ]
                 if y is not None:
                     y = y[self.start:self.stop]
 
-            return X, y
+            return x, y
+
+        if self.shuffle:
+            np.random.shuffle(data)
```

Comment:
This part is not needed, we can simply use "shuffled_sequential" as a mode to iterate over a shuffled version.
Also, it is problematic because it is not reproducible, we cannot control which permutation will be applied.
So I would get rid of the "shuffle" option.